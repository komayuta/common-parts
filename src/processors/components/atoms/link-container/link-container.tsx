import * as React from "react";

import {Link} from "../link/link";

export const LinkContainer: React.FC<React.AnchorHTMLAttributes<HTMLAnchorElement>> = (props) => {
    const {href, children} = props;
    return href ? <Link {...props} /> : <>{children}</>;
};
