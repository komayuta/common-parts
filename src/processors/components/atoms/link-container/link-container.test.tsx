import {render} from "@testing-library/react";
import * as React from "react";

import {LinkContainer} from "./link-container";

describe("LinkContainer", () => {
    test("renders <a> if href exists", () => {
        // act
        const element = render(
            <LinkContainer href={"/abc"} className={"test-class"}>
                <b>test</b>
            </LinkContainer>,
        ).container.firstChild as Element;

        // assert
        expect(element.tagName.toLowerCase()).toBe("a");
        expect(element).toContainHTML("<b>test</b>");
        expect(element).toHaveAttribute("class", "test-class");
        expect(element).toHaveAttribute("href", "/abc");

        expect(element).toMatchSnapshot();
    });

    test("does not render <a> if href is invalid", () => {
        // act
        const element = render(
            <LinkContainer href={""} className={"test-class"}>
                <b>test</b>
            </LinkContainer>,
        ).container.firstChild as Element;

        // assert
        expect(element.tagName.toLowerCase()).toBe("b");
        expect(element).toContainHTML("test");

        expect(element).toMatchSnapshot();
    });
});
