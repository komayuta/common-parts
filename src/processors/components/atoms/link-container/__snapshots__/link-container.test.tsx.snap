// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkContainer does not render <a> if href is invalid 1`] = `
<b>
  test
</b>
`;

exports[`LinkContainer renders <a> if href exists 1`] = `
<a
  class="test-class"
  href="/abc"
>
  <b>
    test
  </b>
</a>
`;
